### ISSUE TITLE
The "quick inserter" button is hardly usable or impossible to use with a keyboard.

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#24862]
**Describe the bug**

The "quick inserter" button is displayed in the UI in different ways, depending on the block state (empty), number of blocks, etc. In all of these cases there's a constant though: keyboard accessibility is far from ideal, at a point that in some cases this button can't be used with a keyboard.

So far, I was able to identify two main issues:
- the placement of the button in the DOM is far from ideal: visual/reading order and DOM order mismatch, so that the tab sequence is unexpected and very confusing for users
- sometimes, the button is only revealed on mouse hover so that it can't be used by keyboard users

**First Example:** 
In the empty block, the quick inserter is visually displayed on the right of the empty block:

<img width="1003" alt="Screenshot 2020-08-26 at 11 27 07" src="https://user-images.githubusercontent.com/1682452/91445740-c9f0ae00-e876-11ea-8ab1-9a5d4981e708.png">

Since it's visually displayed _after_ the block editable area, I'd expect the tab sequence to go first to the editable area and then to the quick inserter button. However, the button is placed in the DOM _before_ the editable area thus making it hard to use with a keyboard. To test:
- create a new post
- enter something in the title
- press Tab
- the empty block receives focus and the quick inserter button appears on the right
- press Tab again
- the quick inserter button is skipped: focus goes to the sidebar
- press Shift+Tab to tab through the content backwards
- from the sidebar, focus goes first to the empty block editable area
- press Tab again: focus goes to the quick inserter button

Recap:
- the quick inserter button can't be used when tabbing forwards
- it can only be used when tabbing backwards, which is definitely counterintuitive 
- blind screen reader users may completely miss the quick inserter button is there because it gets skipped when normally navigating content forwards

**Second example:**
In other block types, for example an Image block, the quick inserter button seem to appear only on mouse hover. Or, at least, I couldn't find a way to make it appear and use it with the keyboard. To test:
- edit a post that contains an Image block
- navigate to the Image block using the keyboard
- Tab and Shift+Tab around the Image block in both "Navigation" or "Edit" mode
- the quick inserter button doesn't appear
  - note: the button _before_ the block does appear, but the one _after_ the block doesn't

Additionally:
- for testing purposes mouse hover the area below the Image block so that the button appears:

<img width="1000" alt="Screenshot 2020-08-26 at 11 47 45" src="https://user-images.githubusercontent.com/1682452/91447782-59975c00-e879-11ea-8638-f8908e32fdca.png">

- start tabbing from the block toolbar
- press Tab: focus goes to the quick inserter button, skipping the block

<img width="996" alt="Screenshot 2020-08-26 at 11 47 49" src="https://user-images.githubusercontent.com/1682452/91448787-9c0d6880-e87a-11ea-88c1-990d05f3bf2b.png">

- press Tab: focus goes back to the block

<img width="999" alt="Screenshot 2020-08-26 at 11 47 54" src="https://user-images.githubusercontent.com/1682452/91448804-a3347680-e87a-11ea-83bb-c6b2052f633d.png">

To recap:
- the quick inserter button visually placed after the block can't be used with a keyboard
- even when the button is visible because hovered with the mouse, the visual order and DOM order mismatch thus resulting in an unexpected tab sequence that is very confusing for users

Worth reminding that for accessibility the visual/reading order and the DOM order must match when they affect meaning and operation. This is a fundamental principle that needs to be always implemented in any accessible UI.

WCAG 2.1 references:
- [1.3.2 Meaningful Sequence (Level A)](https://www.w3.org/TR/WCAG21/#meaningful-sequence) 
- [2.4.3 Focus Order (Level A)](https://www.w3.org/TR/WCAG21/#focus-order)
